### ISSUE TITLE
Do I need to convert data to device for TensorRTExecutionProvider?

### ISSUE THREAD

**Issue by davodogster** [microsoft/onnxruntime#13952]
### Describe the issue

I trained a pytorch lightning model 'arch':'deeplabv3plus', 'encoder_name':"tu-resnetrs50" in SMP. Converted it to ONNX, ran shape inference. I am able to run inference in TRT (data on CPU), But it is only about 30% faster than raw pytorch GPU inference.

The Python API Docs only discuss converting to device for CUDA+CPU Execution provider but nothing for TRT. Can we convert to cuda device for TRTExecution? Will it speed up inference> How is it done?

Cheers, Sam

### To reproduce

![Uploading image.png…]()

### Urgency

moderate

### Platform

Linux

### OS Version

Ubuntu 20.04

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

1.13.1

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

TensorRT

### Execution Provider Library Version

CUDA 11.6 TRT 8.4.1.5

**Comment by faxu** [comment]
What do you mean by "convert data to device"? TRT EP must be registered to be used, see https://onnxruntime.ai/docs/execution-providers/TensorRT-ExecutionProvider.html#python

**Comment by davodogster** [comment]
@faxu Yes I know about registering TRT EP. I mean to("cuda"). DEVICE = "cuda", thanks

**Comment by davodogster** [comment]
> What do you mean by "convert data to device"? TRT EP must be registered to be used, see https://onnxruntime.ai/docs/execution-providers/TensorRT-ExecutionProvider.html#python

![image](https://user-images.githubusercontent.com/46079516/215386926-7f021622-e397-4380-aa43-edc031e638d8.png)

Can we convert data to cuda before passing to TRT EP ? If we can, how do we do it? Thanks

**Comment by davodogster** [comment]
@faxu 

Here is what Im trying to do:

![image](https://user-images.githubusercontent.com/46079516/215938023-b01603d0-7d30-495d-b76c-bb60086dbc9e.png)

![image](https://user-images.githubusercontent.com/46079516/215938141-ba13fa66-a5ac-471a-b9b1-c8ca71e66194.png)

![image](https://user-images.githubusercontent.com/46079516/215938193-4b59266d-39f9-4a57-9715-24e6bf3aab82.png)

Thanks, Sam

**Comment by hariharans29** [comment]
There are several parts to your question:

1) The error you see above is because you are trying to bind `None` as an output and this is not a valid usage of the API. You would have to use the right output name and optionally provide a buffer that will be used as the buffer where the output will be written to. See all illustrative examples in the IOBinding doc you referenced above. 

2) I think your actual question is : "Can we use IOBinding for the TensorRT EP and will that provide a copy savings when compared to feeding in data from the CPU for a TensorRT EP backed session ?". For this, I am tagging the TensorRT EP folks - @jywu-msft @stevenlix

**Comment by davodogster** [comment]
@hariharans29 Ok do you know how I get the output name? And yes to #2! That is my question. Thanks, Sam
